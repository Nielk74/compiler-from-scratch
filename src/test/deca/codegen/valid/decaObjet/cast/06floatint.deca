// Description:
//    test qui cast un float en int (arrondi le float Ã  l'entier le plus proche de 0)
//
// Resultats:
//    7
//    2147483647
//
// Historique:
//    cree le 21/06/2022

{
    float x = 2.6;
    int y = (int)(x * x + x - 2);
    println(y);
    println((int)(0x1.0p120));
}