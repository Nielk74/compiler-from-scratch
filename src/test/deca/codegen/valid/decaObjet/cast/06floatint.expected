7
2147483647
